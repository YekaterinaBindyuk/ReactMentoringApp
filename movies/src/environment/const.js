export const BASE_IMG_CLASSNAME = 'img';
export const MOVIE_LIST_ITEM_COVER = 'small';
export const MOVIE_COVER = 'large';
export const MAIN_LOGO = 'http://localhost:8080/src/images/logo.png';
export const SEARCH_BUTTON = 'Search';
export const TITLE_TYPE_BUTTON = 'title';
export const GENRE_TYPE_BUTTON = 'genre';
export const WARNING = 'Something went wrong';
export const RATING_TYPE_BUTTON = 'rating';
export const RELEASE_TYPE_BUTTON = 'release date';
export const SORTING_TEXT = 'Sort by: ';
export const SEARCH_TEXT = 'Search by: ';
export const NO_FILMS_WARNING = 'No films found';
export const MOVIES_COUNT_TEXT = ' films found:';
export const RECOMMENDED_MOVIES = 'Recommended movies';
export const MAIN_PAGE = '/';
export const REST_URL = 'https://reactjs-cdp.herokuapp.com/';
export const MOVIES_URL = REST_URL + 'movies';
