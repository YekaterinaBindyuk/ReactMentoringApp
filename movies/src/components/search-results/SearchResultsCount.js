import React from 'react';
import {MOVIES_COUNT_TEXT} from '../../environment/const';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const SearchResultsCount = (props) => {
    const {count} = props;
    return(
        <div className="movies-count">
        <div className="badge badge-dark p-2"><FontAwesomeIcon icon="film" className="mlr-1"/><span>{MOVIES_COUNT_TEXT} <span className="badge badge-light">{count}</span></span></div>
        </div>
    )
}

export default SearchResultsCount;