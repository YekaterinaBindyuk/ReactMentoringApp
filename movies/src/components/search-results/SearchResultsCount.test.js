import React from "react";
import { render, shallow } from "enzyme";
import { MOVIES_COUNT_TEXT } from "../../environment/const";

import SearchResultsCount from "./SearchResultsCount";
describe("SearchResults", () => {
  it('should render correctly in "debug" mode', () => {
    const component = render(<SearchResultsCount debug count={1} />);
    expect(component).toMatchSnapshot();
  });
});

it("renders movies count", () => {
  const count = 1;
  const wrapper = shallow(<SearchResultsCount count={count} />);
  expect(wrapper.html()).toEqual(
    '<div class="movies-count"><div class="badge badge-dark p-2"><i class="fa"></i><span>' +
      MOVIES_COUNT_TEXT +
      ' <span class="badge badge-light">' +
      count +
      "</span></span></div></div>"
  );
});
