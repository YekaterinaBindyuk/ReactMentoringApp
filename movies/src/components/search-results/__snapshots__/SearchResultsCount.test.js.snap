// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should render correctly in "debug" mode 1`] = `
<div
  class="movies-count"
>
  <div
    class="badge badge-dark p-2"
  >
    <i
      class="fa"
    />
    <span>
       films found: 
      <span
        class="badge badge-light"
      >
        1
      </span>
    </span>
  </div>
</div>
`;
